src/player.cpp

```diff
@@ -9610,32 +9610,41 @@ bool player::wield( item& target )
 
     int mv = 0;
 
+    bool target_removed = false;
+    item target_copy = target;
     if( is_armed() ) {
+        inventory old_inv = inv;
```

Comment:
You shouldn't call `i_rem` until it is confirmed that the item will be wielded.

`i_rem` is intended to cover side effects from item removal, meaning it shouldn't happen when just trying to do something.

A more correct way of doing this would be to:
* Get current capacity
* Subtract current inventory volume
* Add held item volume
* Subtract to-be-wielded item volume
* Check if `>0`